tests/contrib/httplib/test_httplib.py

```diff
@@ -0,0 +1,360 @@
+            with contextlib.closing(conn):
+                conn.request('GET', '/500')
+                conn.getresponse()
+        except httplib.HTTPException:
+            resp = sys.exc_info()[1]
+            self.assertEqual(self.to_str(resp.read()), '500 Internal Server Error')
+            self.assertEqual(resp.status, 500)
+
+        spans = self.tracer.writer.pop()
+        self.assertEqual(len(spans), 1)
+        span = spans[0]
+        self.assertEqual(span.span_type, 'http')
+        self.assertIsNone(span.service)
+        self.assertEqual(span.name, self.SPAN_NAME)
+        self.assertEqual(span.error, 1)
+        self.assertEqual(span.get_tag('http.method'), 'GET')
+        self.assertEqual(span.get_tag('http.status_code'), '500')
+        self.assertEqual(span.get_tag('http.url'), 'http://httpstat.us/500')
+
+    def test_httplib_request_non_200_request(self):
+        """
+        When making a GET request via httplib.HTTPConnection.request
+            when the response is a non-200
+                we raise the original exception
+                we mark the span as an error
+                we capture the correct span tags
+        """
+        try:
+            conn = httplib.HTTPConnection('httpstat.us')
+            with contextlib.closing(conn):
+                conn.request('GET', '/404')
+                conn.getresponse()
+        except httplib.HTTPException:
+            resp = sys.exc_info()[1]
+            self.assertEqual(self.to_str(resp.read()), '404 Not Found')
+            self.assertEqual(resp.status, 404)
+
+        spans = self.tracer.writer.pop()
+        self.assertEqual(len(spans), 1)
+        span = spans[0]
+        self.assertEqual(span.span_type, 'http')
+        self.assertIsNone(span.service)
+        self.assertEqual(span.name, self.SPAN_NAME)
+        self.assertEqual(span.error, 0)
+        self.assertEqual(span.get_tag('http.method'), 'GET')
+        self.assertEqual(span.get_tag('http.status_code'), '404')
+        self.assertEqual(span.get_tag('http.url'), 'http://httpstat.us/404')
+
+    def test_httplib_request_get_request_disabled(self):
+        """
+        When making a GET request via httplib.HTTPConnection.request
+            when the tracer is disabled
+                we do not capture any spans
+        """
+        self.tracer.enabled = False
+        conn = httplib.HTTPConnection('httpstat.us')
+        with contextlib.closing(conn):
+            conn.request('GET', '/200')
+            resp = conn.getresponse()
+            self.assertEqual(self.to_str(resp.read()), '200 OK')
+            self.assertEqual(resp.status, 200)
+
+        spans = self.tracer.writer.pop()
+        self.assertEqual(len(spans), 0)
+
+    def test_urllib_request(self):
+        """
+        When making a request via urllib.request.urlopen
+           we return the original response
+           we capture a span for the request
+        """
+        resp = urlopen('http://httpstat.us/200')
```

Comment:
@palazzem I am having trouble updating these tests here to support attaching a `Pin` to each `httplib.HTTPConnection` instance. The problem is I don't seem to have direct access to the `HTTPConnection` instance before the request is made.

Any thoughts/suggestions here?

For the other tests I am doing:

```
request = httplib.HTTPConnection(*args, **kwargs)
Pin.override(request, tracer=self.tracer)
```